src/gallium/drivers/lima/lima_draw.c

```diff
@@ -908,6 +916,31 @@ lima_draw_vbo(struct pipe_context *pctx, const struct pipe_draw_info *info)
       ctx->const_buffer[PIPE_SHADER_FRAGMENT].dirty = false;
    }
 
+   if (!ctx->num_draws ||
+      (ctx->dirty & LIMA_CONTEXT_DIRTY_TEXTURES)) {
+      lima_update_textures(ctx);
+      if (lima_dump_command_stream) {
```

Comment:
Maybe it's better to move this dump to lima_update_textures for a clean lima_draw_vbo in which case lima_dump_blob need be moved to lima_util.c too.

Reply:
OK